lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -65,23 +120,122 @@ pub mod aggregator {
-    fn submit_service_check(name: String, status: i32, tags: Vec<String>, hostname: String, message: Option<String>) {
-        println!(
+    fn submit_service_check(
+        _class: PyObject, _check_id: String, name: String, status: u8, tags: Vec<String>, hostname: String,
+        message: Option<String>,
+    ) {
+        trace!(
             "submit_service_check called with name: {}, status: {}, tags: {:?}, hostname: {}, message: {:?}",
-            name, status, tags, hostname, message
+            name,
+            status,
+            tags,
+            hostname,
+            message
         );
+
+        if let Ok(service_check_status) = status.try_into() {
+            let mut tag_set = TagSet::default();
+            for tag in tags {
+                tag_set.insert_tag(tag);
+            }
+
+            let service_check = ServiceCheck::new(name, service_check_status)
+                .with_tags(tag_set.into_shared())
+                .with_hostname(if hostname.is_empty() {
+                    None
+                } else {
+                    Some(hostname.into())
+                })
+                .with_message(message.map(Into::into));
+
+            if let Err(e) = try_send_service_check(service_check) {
+                error!("Failed to send service check: {}", e);
+            }
+        } else {
+            error!("Invalid service check status: {}", status);
+        }
     }
 
-    #[pyfunction]
-    fn metrics(name: String) -> Vec<String> {
-        println!("metrics called for: {}", name);
-        vec![] // Dummy return
+    #[derive(FromPyObject)]
+    struct PythonEvent {
+        #[pyo3(item("msg_title"))]
+        msg_title: String,
+        #[pyo3(item("msg_text"))]
+        msg_text: String,
+        #[pyo3(item("event_type"), default)]
+        #[allow(dead_code)]
+        event_type: Option<String>,
+        #[pyo3(item("timestamp"), default)]
+        timestamp: Option<u64>,
+        #[pyo3(item("api_key"), default)]
+        #[allow(dead_code)]
+        api_key: Option<String>,
+        #[pyo3(item("aggregation_key"), default)]
+        aggregation_key: Option<String>,
+        #[pyo3(item("alert_type"), default)]
+        alert_type: Option<String>,
+        #[pyo3(item("source_type_name"), default)]
+        source_type_name: Option<String>,
+        #[pyo3(item("host"), default)]
+        host: Option<String>,
+        #[pyo3(item("tags"), default)]
+        tags: Option<Vec<String>>,
+        #[pyo3(item("priority"), default)]
+        priority: Option<String>,
     }
 
     #[pyfunction]
-    fn reset() {
-        println!("reset called");
+    fn submit_event(_class: PyObject, _check_id: String, event: PyObject) {
+        let extracted_event = Python::with_gil(|py| -> Result<PythonEvent, PyErr> {
+            let python_event = event.downcast_bound::<PyDict>(py)?;
+            let extracted_event: PythonEvent = python_event.extract()?;
+            Ok(extracted_event)
```

Comment:
Does this still compile if you change these two lines to just be `python_event.extract()`? You're already declaring the return type for the closure so I would expect it to be able to infer the types.

Reply:
I tried, and it does not compile :(